addons/website_hr_recruitment/controllers/main.py

```diff
@@ -277,3 +278,13 @@ def jobs_compatibility(self, country=None, department=None, office_id=None, cont
             contract_type_id=contract_type_id,
             **kwargs
         )
+
+    @http.route('/website_hr_recruitment/check_recent_application', type='json', auth="public")
+    def check_recent_application(self, email, job_id):
+        date_limit = datetime.now() - timedelta(days=90)
+        domain = [('email_from', '=ilike', email),
+                  ('create_date', '>=', date_limit)]
+        recent_applications = http.request.env['hr.applicant'].sudo().search(domain)
+        response = {'applied_same_job': any(recent_applications.filtered(lambda a: a.job_id.id == int(job_id))),
+                    'applied_other_job': any(recent_applications)}
```

Comment:
job_id should be int, no? so we don't need to convert in.. 
also, filtered has linear complexity.. any as well..  what you had originally, will traverse the recent_pplications at least once. This way we will stop as soon as we find te applicant with current job
for the applied applied_other_job, I guess bool is more appropriate
```suggestion
        response = {'applied_same_job': any(a.job_id.id == job_id for a in recent_applications)),
                    'applied_other_job': bool(recent_applications)}
```